katharsis-client/src/main/java/io/katharsis/client/KatharsisClient.java

```diff
@@ -166,6 +166,21 @@ public RelationshipRepository buildRepository() {
 		ResourceRepositoryAdapter repositoryAdapter = entry.getResourceRepository(null);
 		return (ResourceRepositoryStub<T, ID>) repositoryAdapter.getResourceRepository();
 	}
+	
+	/**
+	 * @param resourceClass resource class
+	 * @return stub for the given resourceClass
+	 */
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public <T, ID extends Serializable> QuerySpecResourceRepositoryStub<T, ID> getQuerySpecRepository(Class<T> resourceClass) {
+		init();
+
+		RegistryEntry<T> entry = resourceRegistry.getEntry(resourceClass);
+
+		// TODO fix this in katharsis, should be able to get original resource
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)